### PR TITLE
Add a process for becoming a maintainer

### PR DESCRIPTION
**Pull request by dyladan**
As discussed last week, here is a draft of a process for electing new SIG maintainers.

### CODE REVIEW

File: community-membership.md
Comment (yurishkuro) [inline community-membership.md:231]:
```suggestion

#### Self-nomination is possible and encouraged

```

File: community-membership.md
Comment (yurishkuro) [inline community-membership.md:240]:
```suggestion
of additional maintainers, they may suggest an alternate role or growth areas 
to improve your chances to become a maintainer in the future.
```

File: community-membership.md
Comment (jpkrohling) [inline community-membership.md:236]: Should we list the common expectations connected to this role? Like:

- performing releases
- answering users questions
- fixing bugs across the entire code base
- mentoring other prospective maintainers

Comment (dyladan) [reply community-membership.md:236]: Isn't that listed above in requrements/responsibilities/privileges sections?

File: community-membership.md
Comment (yurishkuro) [inline community-membership.md:225]: I don't have a strong opinion about the "unanimous vote" part, but for the reference the Jaeger's process is a bit more relaxed:

> Two other maintainers need to second the nomination. If no one objects in 5 working days (U.S.), the nomination is accepted. If anyone objects or wants more information, the maintainers discuss and usually come to a consensus (within the 5 working days). If issues can't be resolved, there's a simple majority vote among current maintainers.

File: community-membership.md
Comment (yurishkuro) [inline community-membership.md:232]: > requirements above

I don't think "above" actually lists any requirements to _becoming a maintainer_, it only lists responsibilities of _being_ a maintainer. In Jaeger we had [specific requirements][1] that must be met, where the actual required stats were less important than the fact that they are written down (for a fair process).

Do we want to propose an org-level maintainer stats requirements so that SIGs won't need to define their own (but still have the option to do so)?

[1]: https://github.com/jaegertracing/jaeger/blob/main/GOVERNANCE.md#becoming-a-maintainer

File: community-membership.md
Comment (dyladan) [reply community-membership.md:225]: Our project is broken into many small groups who each maintain only one portion with only a few maintainers. Unanimous consent of one SIG represents only a small portion of the overall maintainers and shouldn't be too high of a bar I think.

File: community-membership.md
Comment (dyladan) [reply community-membership.md:232]: https://github.com/open-telemetry/community/blob/main/community-membership.md#requirements-3

Comment (yurishkuro) [reply community-membership.md:234]: It's not qualitative, and in fact the only line related to direct code contributions (which I would consider a must for all non-spec repos) is somewhat weak:

> Directly contributed to the subproject through implementation and / or review

Comment (dyladan) [reply community-membership.md:234]: i think the requirements of the position should be a separate PR from the process itself

Comment (yurishkuro) [reply community-membership.md:234]: +1

File: community-membership.md
Comment (SergeyKanzhelev) [reply community-membership.md:225]: I like the lazy consensus part of @yurishkuro 's quote. Active consensus sometimes hard to get.

File: community-membership.md
Comment (bhs) [inline community-membership.md:183]: nit: are we calling these people Owners or Maintainers? (I prefer Maintainers since we've already used that term above) it seems (?) like we're using those terms interchangeably here.

File: community-membership.md
Comment (bhs) [inline community-membership.md:187]: s/owner/maintainer/ (per the above)?

File: community-membership.md
Comment (bhs) [inline community-membership.md:244]: Given the requirement of unanimity above, I'd suggest that we augment this section to include "non-technical conflicts" (i.e., have some escalation process for removing someone as a maintainer in a SIG).

... Or, alternatively, reduce the requirement for adding a maintainer to be less than a unanimous vote of existing maintainers?

File: community-membership.md
Comment (dyladan) [reply community-membership.md:225]: For most things I'd agree but it seems like adding a maintainer is about the most impactful thing you can do. If a maintainer can't be bothered to consent to a new maintainer either the new maintainer isn't agreed on or the existing maintainer isn't doing their job imo

File: community-membership.md
Comment (jpkrohling) [reply community-membership.md:238]: Sorry about that, it indeed is!

File: community-membership.md
Comment (jpkrohling) [reply community-membership.md:225]: > or the existing maintainer isn't doing their job

This might indeed happen: maintainers who have moved on might not have yet turned themselves into emeritus and might be hard to reach. Same for maintainers on longer breaks.

Comment (dyladan) [reply community-membership.md:225]: So how about if there is a unanimous vote then the vote passes, otherwise if there is no objection within 5 days the vote passes?

File: community-membership.md
Comment (dyladan) [reply community-membership.md:183]: I took the wording from the "requirements" section above. I'll change them both.

File: community-membership.md
Comment (dyladan) [reply community-membership.md:248]: It seems others feel the same way about the unanimous vote so I loosened the restrictions. PTAL.

File: community-membership.md
Comment (dyladan) [reply community-membership.md]: I updated the PR to reflect this lazy consensus. PTAL

**Review by jpkrohling — APPROVED** [review]

**Review by yurishkuro — APPROVED** [review]

File: community-membership.md
Comment (tigrannajaryan) [inline community-membership.md:231]: I am not sure I understand the "OR" here. It seems like the right-side of the OR is a sufficient condition and the left-side is superfluous.

**Review by tigrannajaryan — APPROVED** [review]

File: community-membership.md
Comment (dyladan) [reply community-membership.md:231]: if all maintainers approve you don't need to wait the full 5 days. should we just wait 5 days no matter what?

Comment (tigrannajaryan) [reply community-membership.md:231]: Ah, I see, there is a difference then. I think what you have is fine.

**Review by SergeyKanzhelev — APPROVED** [review]

**Comment by dyladan** [comment]
I think i've addressed all comments. If I haven't gotten any objections I'll merge this monday morning.

**Comment by dyladan** [comment]
@bhs you're the only person who made a comment but has not yet approved. Want to give it one last look before merge?

**Review by cijothomas — APPROVED** [review]

**Review by MrAlias — APPROVED** [review]

**Review by bogdandrutu — APPROVED** [review]

File: community-membership.md
Comment (dyladan) [reply community-membership.md]: You're right. Added 3e035710659d76dab81631a3bfcd8b3716fa5316

**Review by bhs — APPROVED** [review]
Sorry about that, @dyladan ! Your changes looked (and look) good to me.

**Review by carlosalberto — APPROVED** [review]

**Review by Aneurysm9 — APPROVED** [review]

**Review by mtwo — APPROVED** [review]

**Review by alolita — APPROVED** [review]
+1

**Pull request merged**
